saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/Xml/NlpXmlReader.md

```diff
@@ -0,0 +1,115 @@
+# NLP Xml Reader
+An [Xml reader](NlpXmlReader.java) that facilitates reading data from xml files into Saul's NLP 
+[BaseTypes](../BaseTypes/BaseTyps.md).
+
+##Tag Names
+The reader uses `"DOCUMENT"`, `"SENTENCE"`, `"PHRASE"`, and `"TOKEN"` as default tag names 
+for the corresponding base types. You can change them either by the constructor or by the
+setters provided in the reader's class. Note that you can read a type from multiple tags by
+changing the tag name just before reading the data.
+
+The properties or tags that determine `start`, `end`, `id` and `text` of each constituent
+are predefined in the reader and can be changed using the corresponding setters as well.
+
+##Read a list of constituents
+Using `getDocuments`, `getSentences`, `getPhrases` and `getTokens` we can retrieve any of 
+the base types in the xml file. For each type except the document there is another function
+`getXXXByParentId` which retrieves list of a constituent which contained in another (Parent)constituent.
+
+`getRelations` and `getRelationsByParentId` work similar to aforementioned functions, but need
+to determine the property name of argument ids of the relation as well.
+
+## Adding properties to the constituents from other tags
+In many scenarios we need to add properties from various tags. We can do that by providing
+the tag names when using `getXXXs` or `getXXXByParentId` when we have the constituents in 
+the xml file. But when we have generated the constituents from another source and want to 
+add properties from a tag, we can use `addPropertiesFromTag`. This function uses the parent Id
+of the constituents(`documentId` for sentence, and `sentenceId` for `Token` and `Phrase`) to find
+the context for each constituent to be retrieved.
+
+### Matching
+adding properties from a tag requires matching between constituents and tags. The default
+matching strategy is [`ExactMatching`](XmlExachMatching.java). 
+
+Built in matching strategies:
+- [`ExactMatching`](XmlExachMatching.java): adds tag's properties if the tag's span
+ exactly matches with the constituent's span 
+- [`InclusionMatching`](XmlInclusionMatching.java): adds tag's properties if the tag's span
+ includes the constituent's span
+- [`PartOfMatching`](XmlPartOfMatching.java): adds tag's properties if the tag's span is
+a part of constituent span, in other word if the constituent's span includes the tag's span
+- [`OverlapMatching`](XmlOverlapMatching.java): adds tag's properties if the tag's span
+and the constituent's span are overlapping
+- [`HeadwordMatching`](../../../../../../../../scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/XmlMatchings.scala):
+ adds tag's properties if the tag's span contains the headwords span of the constituent
+
```

Comment:
also one or two small example how these similarity functions are used, when reading the xml